scripts/apps/authoring/views/authoring-topbar.html

```diff
@@ -58,28 +58,25 @@
                 <span class="btn__text--short" translate>T D</span>
         </button>
         <button
-                ng-if="$root.config.features.customAuthoringTopbar.closeAndContinue === true"
+                ng-if="$root.config.features.customAuthoringTopbar.closeAndContinue === true && canRewriteArticle()"
                 class="btn btn--custom"
-                ng-disabled="!canRewriteArticle()"
                 ng-click="saveAndContinue(closeAndContinue, false)"
                 title="{{gettext('Close and Continue')}}">
                 <span class="btn__text" translate>Close and Continue</span>
                 <span class="btn__text--short" translate>C & C</span>
         </button>
         <button
-                ng-if="$root.config.features.customAuthoringTopbar.publish === true"
+                ng-if="$root.config.features.customAuthoringTopbar.publish === true && canPublishOnDesk()"
                 class="btn btn--custom btn--publish"
                 type="submit"
-                ng-if="canPublishOnDesk()"
                 ng-click="saveAndContinue(publish, true)"
                 title="{{ :: 'Publish' | translate }}">
                 <span class="btn__text" translate>Publish</span>
                 <span class="btn__text--short" translate>P</span>
         </button>
         <button
-                ng-if="$root.config.features.customAuthoringTopbar.publishAndContinue === true"
+                ng-if="$root.config.features.customAuthoringTopbar.publishAndContinue === true && canPublishOnDesk() && canRewriteArticle()"
```

Comment:
Why is `canRewriteArticle` checked for `Publish and continue` action, but not for `Publish`?

Reply:
`Publish and continue` publishes a story and creates an update right after, That's why it requires `canRewriteArticle` permission.